ctrlf.el

```diff
@@ -1015,6 +1067,61 @@ search, change back to regexp search."
       (isearch-backward-regexp)
     (ctrlf-backward 'regexp)))
 
+;;;###autoload
+(defun ctrlf-forward-symbol ()
+  "Search forward for symbol.
+If already in a search, go to next candidate, or if no input then
+insert the previous search string. If in a non-symbol search,
+change back to symbol search."
+  (interactive)
+  (if (and (window-minibuffer-p) (not ctrlf--active-p))
+      (isearch-forward-symbol)
+    (ctrlf-forward 'symbol)))
+
+;;;###autoload
+(defun ctrlf-forward-symbol-at-point ()
+  "Search forward for symbol at point.
+If already in a search, replace the current input and change to a
+symbol search, otherwise start the search. If no symbol is found,
+display an error message and do not search."
+  (interactive)
+  (let ((arg nil)
+        (pos nil)
+        (skip-search nil))
+    (with-current-buffer (window-buffer (minibuffer-selected-window))
+      (setq pos (point)) ; Save original point.
+      ;; NOTE: This idea is borrowed from `isearch' but I don't see
+      ;; how else it could be implemented.
+      (let ((start (car (find-tag-default-bounds)))) ; Find start of symbol.
+        (when (and start (< start (point))) ; `start' can be nil.
+          (setq ctrlf--current-starting-point start) ; For `ctrlf-forward'.
+          (goto-char start))) ; Side-effect! Hence saving point above.
+      (setq arg (thing-at-point 'symbol t))
+      (unless arg
+        (setq skip-search t)
+        (let ((ctrlf--message-persist-p t))
+          (ctrlf--message "No symbol at point"))))
+    (unless skip-search
+      ;; NOTE: We pass `pos' so that `ctrlf-cancel' returns to the
+      ;; original position, and not the start of the symbol at point.
+      (ctrlf-forward 'symbol nil arg pos))))
```

Comment:
Would it work if in `ctrlf--start` you set `ctrlf--current-starting-point` to `pos` instead of `ctrlf--starting-point`? Then you might not have to move point in this function, and hence not have to save its original value.

Reply:
I feel like I tried that, but let me try it out again.

Reply:
Ah ha, that worked nicely!